### PR TITLE
feat(shell): guide accidental "sqly help"/"sqly version" to the right flag

### LINKED ISSUE #664: [UX] explain that sqly is flag-driven and guide users away from positional help/version
## Summary
sqly is a flag-driven CLI, but the top-level help does not clearly say that there are no subcommands. A common first try like `sqly help` is interpreted as an input path and fails with an import error.

## Reproduction
```bash
sqly help
sqly version
```

## Actual behavior
The command treats `help` / `version` as file paths and returns errors such as:

```text
Import failed with 1 error(s):
  - path does not exist: help
all import attempts failed
```

## Expected behavior
- The onboarding docs and top-level help should make it obvious that sqly uses flags (`--help`, `--version`) rather than subcommands.
- Ideally, the CLI should also detect the most common accidental positional forms (`help`, `version`) and point the user to the right flag.

## Impact
- New users can conclude the CLI is broken before they reach the interactive shell or query path.
- This is especially confusing because many other CLIs support `tool help` / `tool version`.

## Suggested fix
- Add an explicit note in README and `--help` that sqly has no top-level subcommands and that helper commands live inside the shell / batch stdin mode.
- Consider special-casing `help` and `version` as positional hints that print a short correction instead of an import-path error.

## Suggested regression coverage
- Add a CLI/spec test that asserts `sqly help` exits non-zero with a hint to use `--help` (if behavior is improved).
- Add a docs-focused test or README example check that preserves the new explanation about flag-driven usage and shell helper commands.

### PR DESCRIPTION
**Pull request by nao1215**
## Summary

sqly is flag-driven and has no subcommands, but a common first try like `sqly help` or `sqly version` was parsed as an input path and failed with a confusing import error:

```text
Import failed with 1 error(s):
  - path does not exist: help
all import attempts failed
```

sqly now detects the accidental `help`/`version` positional (case-insensitive) and prints a short hint pointing at `--help`/`--version`, exiting non-zero. A real file or directory named `help`/`version` still imports. The `--help` text, README, and GitHub Pages docs also state that sqly has no subcommands and that helper commands run inside the shell or batch stdin mode.

## Changes

- `shell/shell.go`: add `positionalSubcommandHint`; emit it before import in `Run`.
- `config/argument.go`: add a flag-driven note to the `--help` usage text.
- `README.md`, `doc/pages/markdown/how_to_use.md`: document that sqly has no subcommands.
- `shell/shell_test.go`: `TestPositionalSubcommandHint` (help/version/case-insensitive/real-file-wins/normal-arg/empty).
- `spec/v0_25_bugs_spec.sh`: binary-level cases for `sqly help` and `sqly version`.
- regenerated `config/testdata/usage.golden` and `shell/testdata/golden/help.golden`.

## Test

- `go test ./...`
- `shellspec --shell sh` (357 examples, 0 failures)

Closes #664

### CODE REVIEW

File: shell/shell.go [hunk 1]
```diff
@@ -442,6 +450,32 @@ func (s *Shell) startsInteractiveShell() bool {
...
+// fires only when no such path exists. The match is case-insensitive to also
+// catch "HELP"/"Version".
+func positionalSubcommandHint(paths []string) (string, bool) {
+	if len(paths) == 0 {
+		return "", false
+	}
+	first := paths[0]
+	var flag string
+	switch strings.ToLower(first) {
+	case "help":
```
[the remark this answers is not in the document]
Comment (nao1215) [reply shell/shell.go:466]: Not extracting a constant here. In non-test code each of these strings occurs exactly once (the others reviewdog counts are in this PR test file). The repo .golangci.yml intentionally excludes goconst for *_test.go, and the authoritative golangci-lint CI job is green; reviewdog re-reports without that test exclusion. A single-use constant would not improve readability.

File: shell/shell.go [hunk 2]
```diff
@@ -442,6 +450,32 @@ func (s *Shell) startsInteractiveShell() bool {
...
+		flag = "--help"
```
[the remark this answers is not in the document]
Comment (nao1215) [reply shell/shell.go:467]: Not extracting a constant here. In non-test code each of these strings occurs exactly once (the others reviewdog counts are in this PR test file). The repo .golangci.yml intentionally excludes goconst for *_test.go, and the authoritative golangci-lint CI job is green; reviewdog re-reports without that test exclusion. A single-use constant would not improve readability.

File: shell/shell.go [hunk 3]
```diff
@@ -442,6 +450,32 @@ func (s *Shell) startsInteractiveShell() bool {
...
+	case "version":
+		flag = "--version"
```
[the remark this answers is not in the document]
Comment (nao1215) [reply shell/shell.go:469]: Not extracting a constant here. In non-test code each of these strings occurs exactly once (the others reviewdog counts are in this PR test file). The repo .golangci.yml intentionally excludes goconst for *_test.go, and the authoritative golangci-lint CI job is green; reviewdog re-reports without that test exclusion. A single-use constant would not improve readability.
